spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/MongoTemplate.java

```diff
@@ -2559,8 +2561,8 @@ protected <T> T doFindOne(String collectionName, CollectionPreparer<MongoCollect
 	 * @return the List of converted objects.
 	 */
 	protected <T> List<T> doFind(String collectionName, CollectionPreparer<MongoCollection<Document>> collectionPreparer,
-			Document query, Document fields, Class<T> entityClass) {
-		return doFind(collectionName, collectionPreparer, query, fields, entityClass, null,
+			Document query, Document fields, Document sort, Class<T> entityClass) {
```

Comment:
We cannot simply change the signature of our API. With sort being an essential part for query logging, we might consider reworking `QueryCursorPreparer` to remove sorting from the preparer and putting it into method signatures.

Reply:
@mp911de - this PR was just to show what is the problem. The easiest way was to show what was incorrectly logged.